CoreFoundation/String.subproj/CFString.c

```diff
@@ -6442,9 +6442,9 @@ reswtch:switch (ch) {
 		return true;
 	    }
 	case '1': case '2': case '3': case '4': case '5': case '6': case '7': case '8': case '9': {
-	    int64_t number = 0;
+            long long number = 0;
 	    do {
-                if (__builtin_smulll_overflow(number, 10, &number) || __builtin_saddll_overflow(number, ch - '0', &number)) {
+                if (__builtin_smulll_overflow(number, 10, &number) || __builtin_saddll_overflow(number, ch - '0', &number) || number > INT64_MAX) {
```

Comment:
Isn't this a tautological comparison?  `number` is a `long long` which is signed.  `INT64_MAX` is signed.  So the value of the condition is always false.

Reply:
`long long` is only defined to be at least 64 bits but could be more if ever ported to other platforms. This is a bit of a nit-picking change but it should make it technically correct.

Reply:
What platform(s) currently define it to something other than 64-bit?

Reply:
I'm okay taking pedantic correctness fixes, since this is code intended to be portable.